ERCS/erc-7599.md

```diff
@@ -0,0 +1,132 @@
+3. Off-chain AI Agent service listens to the `AgentRequested` event and invokes the AI model computation and then receives an execution response.
+4. Off-chain AI Agent service returns the result through `handleAgentResponse`.
+5. 
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY” and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+We define the smart contract interface below:
+
+- **AgentRequested**: This event emits when an AI Agent Account job is requested. The off-chain AI Agent service listens and performs requested tasks, and returns final results.
+- **AbilityURIUpdated**: This event emits when the `abilityURL` is updated.
+- **abilityURI()**: Gets the `abilityURI` for an AI Agent.
+- **setAbilityURI()**: AI Agent Account’s creator/owner can update the `abilityURI`.
+- **requestAgent()**: Passes user input as a function parameter and emits `AgentRequested` event. It can be customized and updated by an AI Agent Account’s creator/owner, such as setting it as a paid request.
+- **handleAgentResponse()**: The result of the call is written through this method, and an index is required to ensure the order of the result.
+
+```solidity
+interface IERC7599 is IERC6551Account {
+    /**
+     * @notice This event emits when an on-chain AI Agent job is requested.
+     */
+    event AgentRequested(uint256 index, bytes input);
+
+    /**
+     * @notice This event emits when the abilityURL is updated.
+     */
+    event AbilityURIUpdated(string uri);
+
+    /**
+     * @notice Gets the abilityURI for an AI Agent.
+     * @return The abilityURI
+     */
+    function abilityURI() external view returns(string memory);
+
+    /**
+     * @notice Update the abilityURI.
+     */
+    function setAbilityURI(string memory uri) external;
+
+    /**
+     * @notice Passes user input as a function parameter and emits AgentReq             uested event. 
+     */
+    function requestAgent(bytes memory input) payable external;
+
+    /**
+    * @notice The result of the call is written through this method, and an index is required to ensure the order of the result.
+     */
+    function handleAgentResponse(uint256 index, bytes response) external;
+}
+```
+
+1. The smart contract implementing IERC7594 MUST implement IERC6551 Account.
+2. The `AbilityURIUpdated` event MUST be emitted when the `abilityURI` is updated.
+3. The `AgentRequested` event MUST be emitted when the `requestAgent` function is requested.
+
+
+## Rationale
+
+The key features of this proposal are:
+
+1. Create a set of standards for AI Agent Account interaction interfaces and processes.
+2. Upgrade AI Agent Account without changing any properties of NFTs by being compatible with the ERC-6551 protocol.
+3. Project parties or developers can easily implement their own AI Agent Accounts, which brings many interesting scenarios.
```

Comment:
The Rationale section should explain technical choices made within your proposal. This text seems to be justifying the proposal as a whole, and so should be in the Motivation section. My favourite analogy is:

> Motivation: We need to build a shed because...
> Rationale: We painted the shed red because...